pkg/crc/network/proxy.go

```diff
@@ -0,0 +1,138 @@
+package network
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+)
+
+var defaultNoProxies = []string{".apps-crc.testing", "127.0.0.1", "api.crc.testing", "localhost"}
+
+// ProxyConfig keeps the proxy configuration for the current environment
+type ProxyConfig struct {
+	httpProxy  string
+	httpsProxy string
+	noProxy    []string
+}
+
+// NewProxyConfig creates a proxy configuration with the specified parameters. If a empty string is passed
+// the corresponding environment variable is checked.
+func NewProxyConfig(httpProxy string, httpsProxy string, noProxy string) (*ProxyConfig, error) {
+	if httpProxy == "" {
+		httpProxy = getProxyFromEnv("http_proxy")
+	}
+	err := ValidateProxyURL(httpProxy)
+	if err != nil {
+		return nil, err
+	}
+
+	if httpsProxy == "" {
+		httpsProxy = getProxyFromEnv("https_proxy")
+	}
+	err = ValidateProxyURL(httpsProxy)
+	if err != nil {
+		return nil, err
+	}
+
+	np := []string{}
+	np = append(np, defaultNoProxies...)
+
+	if noProxy == "" {
+		noProxy = getProxyFromEnv("no_proxy")
+	}
+
+	if noProxy != "" {
+		np = append(np, strings.Split(noProxy, ",")...)
+	}
+
+	config := ProxyConfig{
+		httpProxy:  httpProxy,
+		httpsProxy: httpsProxy,
+		noProxy:    np,
+	}
+
+	return &config, nil
+}
+
+func getProxyFromEnv(proxyScheme string) string {
+	p := os.Getenv(proxyScheme)
+	if p == "" {
+		p = os.Getenv(strings.ToUpper(proxyScheme))
+	}
+	return p
+}
+
+// HttpProxy returns the configured value for the HTTP proxy. The empty string is returned in case HTTP proxy is not set.
+func (p *ProxyConfig) HttpProxy() string {
+	return p.httpProxy
+}
+
+// HttpsProxy returns the configured value for the HTTPS proxy. The empty string is returned in case HTTPS proxy is not set.
+func (p *ProxyConfig) HttpsProxy() string {
+	return p.httpsProxy
+}
+
+// HttpProxy with hidden credentials
+func (p *ProxyConfig) HttpProxyWithHiddenCredentials() string {
```

Comment:
I'd name it `HttpProxyForDisplay()`, it's in my opinion clearer when we want to use it.
I think I would just add a generic `UriStringForDisplay()` and call `UriStringForDisplay(HttpsProxy())` whenever needed. I know Gerard suggested otherwise, so let's keep things this way for now